src/Time/TimeSteppers/TimeStepper.hpp

```diff
@@ -126,13 +128,25 @@ class TimeStepper : public PUP::able {
       const TimeStepId& current_id,
       const TimeDelta& time_step) const noexcept = 0;
 
-  /// Whether a change in the step size is allowed before taking
-  /// a step.
+  /// Whether a change in the step size is allowed during the next step -- this
+  /// is to be evaluated between the `update_u` is called and the time step id
+  /// is advanced.
   template <typename Vars, typename DerivVars>
   bool can_change_step_size(
       const TimeStepId& time_id,
-      const TimeSteppers::History<Vars, DerivVars>& history) const noexcept {
+      const TimeSteppers::History<Vars, DerivVars>& history,
+      bool using_error_measure = false) const noexcept {
     return TimeStepper_detail::fake_virtual_can_change_step_size<
+        creatable_classes>(this, time_id, history, using_error_measure);
+  }
+
+  /// Whether a change in the slab size is permitted -- this is to be evaluated
+  /// immediately after the time step id has been advanced.
+  template <typename Vars, typename DerivVars>
+  bool can_change_slab_size(
```

Comment:
You should never need to forbid a slab size change (except during self-start, but that's special).  What are you trying to do here?  The current behavior is required for observations without dense output.

Reply:
Here I'm just trying to preserve the current behavior in `ChangeSlabSize` -- because in my suggested strategy the `ChangeStepSize` must now happen after the step is evaluated by `update_u` (and the slab is still changed before the update), the time step id is now different for the case when the step can change as compared to when the slab can change.